### PR TITLE
patch: Enhance raft's invalid logic

### PR DESCRIPTION
**Pull request by TCeason**
Now, sometimes the invalid state exists false report:

Hypothesis， we have three nodes: LeaderA, FollowerB, FollowerC

Three nodes are zoned to form 1-1-1 distribution. LeaderA degrade to FollowerA 

recover FollowerB's network, it received a insert but does not execute it, And FollowerB upgrade to LeaderB

Now The Leader has been re-elected and FollowerA's Execute_GTID_SET bigger than LeaderB's Execute_GTID_SET.

### CODE REVIEW

File: src/mysql/api_test.go [hunk 1]
```diff
@@ -303,7 +366,7 @@ func TestGTIDGreaterThan(t *testing.T) {
 	defer db.Close()
 
 	// log
-	log := xlog.NewStdLog(xlog.Level(xlog.PANIC))
+	log := xlog.NewStdLog(xlog.Level(xlog.INFO))
```
Comment (prettyshuang) [inline src/mysql/api_test.go]: should be PANIC

File: src/mysql/api_test.go [hunk 2]
```diff
@@ -178,6 +178,27 @@ func TestWaitUntilAfterGTID(t *testing.T) {
 	assert.Nil(t, err)
 }
 
+func TestGetLocalGTID(t *testing.T) {
+	db, mock, err := sqlmock.New()
+	assert.Nil(t, err)
+	defer db.Close()
+
+	//log
+	log := xlog.NewStdLog(xlog.Level(xlog.INFO))
```
Comment (prettyshuang) [inline src/mysql/api_test.go]: INFO should be PANIC

File: src/raft/follower.go [hunk 3]
```diff
@@ -305,6 +305,7 @@ func (r *Follower) startCheckUpgradeToC() {
 				r.sendClusterPing(respChan)
 				r.resetCheckUpgradeToCTimeout()
 			case rsp := <-respChan:
+				r.fUpgradeToC = false
```
Comment (prettyshuang) [inline src/raft/follower.go]: This causes frequent switching between false and true

File: src/raft/follower.go [hunk 4]
```diff
@@ -315,11 +315,12 @@ func (r *Follower) startCheckUpgradeToC() {
 					}
 				}
 
-				if cnt >= r.GetQuorums() {
+				if cnt < r.GetQuorums() {
 					r.WARNING("ping.responses[%v].more.than.half.upgrade.to.candidate", cnt)
```
Comment (prettyshuang) [inline src/raft/follower.go]: The output does not match the actual meaning, move down to the outside of the block.

**Comment by prettyshuang** [comment]
LGTM

**Comment by BohuTANG** [comment]
LGTM

**Pull request merged**